src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -326,6 +326,7 @@ void ShenandoahControlThread::service_concurrent_normal_cycle(GCCause::Cause cau
   ShenandoahConcurrentGC gc(heap->global_generation(), false);
   if (gc.collect(cause)) {
     // Cycle is complete
+    heap->notify_gc_progress();
```

Comment:
I am trying to understand this. Is the following a reasonable summary of the underlying reasoning?

Any concurrent cycle that completes without degenerating is considered progress irrespective of whether it reclaimed anything. The idea is that if there were allocations during that concurrent cycle, they will have succeeded because, if not we'd have degenerated and not reached here: "collect()" would have returned "false".

Reply:
Right.  The idea is that we completed concurrent GC without a failed allocation, and thus concurrent GC was not cancelled...

Reply:
I suppose we could be more "careful" about how we characterize progress...